src/main/java/com/internship/backend/controller/AuthorityController.java

```diff
@@ -0,0 +1,44 @@
+package com.internship.backend.controller;
+
+import com.internship.backend.model.Authority;
+import com.internship.backend.service.AuthorityService;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.springframework.http.ResponseEntity.notFound;
+import static org.springframework.http.ResponseEntity.ok;
+
+@RestController
+@RequestMapping("api/authority")
+public class AuthorityController {
+
+    @Autowired
+    private AuthorityService authorityService;
+
+    @GetMapping("/getAll")
+    public ResponseEntity<List<Authority>> getAll(){
+        Optional<List<Authority>> users = Optional.ofNullable(authorityService.getAllAuthorities());
```

Comment:
In general, we prefer avoiding optional variables. I think we can call the method directly and check for null or empty values, if needed.

Reply:
Instead of 'Optional<List<Authority>> users = Optional.ofNullable(authorityService.getAllAuthorities());` should I change to  'List<Authority> authority = authorityService.getAllAuthorities();
      if(!authority.isEmpty()){ ... }' ?